server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -134,7 +140,11 @@ protected void clusterManagerOperation(
         final ClusterState unusedState,
         final ActionListener<ClusterHealthResponse> listener
     ) {
-
+        if (request.ensureLocalNodeCommissioned()
+            && discovery instanceof Coordinator
```

Comment:
Should `discovery instanceof Coordinator` be an assertion instead?

Reply:
Asserts wouldn't run on prod. And only coordinator has this node's commission status info. If a developer uses a different Discovery mechanism it might break this. Hence putting this check directly

Reply:
Exactly so tests should fail for a developer, in prod this is expected to be Coordinator

Reply:
Would it be fair to assume that `Coordinator` can only be the `discovery` for all use cases? Can there a plugin which writes there own Discovery model? I see something like this implemented in Gateway service. Here, they assume discovery might not be instance of Coordinator. But I get your point, for this change we can also add asserts and put in if as well to be a little cautious
```
if (discovery instanceof Coordinator) {
            recoveryRunnable = () -> clusterService.submitStateUpdateTask("local-gateway-elected-state", new RecoverStateUpdateTask());
        } else {
            final Gateway gateway = new Gateway(settings, clusterService, listGatewayMetaState);
            recoveryRunnable = () -> gateway.performStateRecovery(new GatewayRecoveryListener());
        }
```